packages/engine/src/physics/classes/PhysicsRapier.ts

```diff
@@ -193,8 +193,9 @@ function createRigidBodyForObject(
 
 function removeCollidersFromRigidBody(entity: Entity, world: World) {
   const rigidBody = getComponent(entity, RigidBodyComponent)
-  for (let index = 0; index < rigidBody.numColliders(); index++) {
-    const collider = rigidBody.collider(index)
+  const numColliders = rigidBody.numColliders()
+  for (let index = 0; index < numColliders; index++) {
+    const collider = rigidBody.collider(0)
```

Comment:
careful here, should be using the index

Reply:
the collider() function indexes into an array whose size is decreased with each removal.
I was actually using the index originally but then updated in this commit (https://github.com/XRFoundation/XREngine/pull/6382/commits/8db8e382585648dd7933464015bf6bb92f01eabb)

Reply:
Ah, cool